### ISSUE TITLE
🐛 BUG: Astro check fails after dependencies update

### ISSUE THREAD

**Issue by eidam** [withastro/language-tools#939]
### Describe the Bug

Ola! 

Looks like [volarjs](https://github.com/volarjs/volar.js) v2.4.0 release is not compatible with `language-tools` and checks are now failing. 

It can be reproduced on any new astro templates, or any project with updated npm lockfile. 

### Steps to Reproduce

1. `npm init astro` (no template)
2. `cd project-name`
3. `npm run build`
4. Error!  `astro check` fails on the following error

```
> blue-belt@0.0.1 build
> astro check && astro build

19:31:41 [types] Generated 29ms
19:31:41 [check] Getting diagnostics for Astro files in /Users/eidam/personal/blue-belt...
Cannot read properties of undefined (reading 'getScriptFileNames')
  Stack trace:
    at AstroCheck.lint (/Users/eidam/personal/blue-belt/node_modules/@astrojs/language-server/dist/check.js:59:84)
    at async check (file:///Users/eidam/personal/blue-belt/node_modules/@astrojs/check/dist/index.js:95:24)
    at async runCommand (file:///Users/eidam/personal/blue-belt/node_modules/astro/dist/cli/index.js:153:27)
```

### Link to Minimal Reproducible Example

https://example.com/any/astro/project

**Comment by justinjunodev** [comment]
Experienced the same issue today as well.

**Comment by Princesseuh** [comment]
The language server wrongfully used `~` deps because of a recent workaround we did for a Yarn bug and the new version of Volar has changed how the check API works (since it was an alpha, this is ok)

I'll release an update asap.

**Comment by Princesseuh** [comment]
This should be fixed in 2.14.1 of the language server

**Comment by RyanHarrigan** [comment]
Found that reverting `@astrojs/check` to `^0.7.0` works for me

**Comment by blackhaj** [comment]
I was getting the same issue with the below versions. Reverting to `^0.7.0` did not work for me.

```
"@astrojs/check": "^0.9.2",
"astro": "4.12.1",
```

However, updating to the latest version of astro did fix it:
```
"@astrojs/check": "^0.9.2",
"astro": "4.14.2",
```

**Comment by eidam** [comment]
I can confirm the issue is solved on the latest (2.14.1) version in my case. Thank you! ❤️

**Comment by killoblanco** [comment]
I did solve it by upgrading `"@astrojs/check"` to version `"0.9.3"`
